### PR TITLE
feat: add customer form to mercury

### PR DESCRIPTION
**Pull request by SamuelBenkeCalabresi**
What's changed
* feat: add customer form to mercury reservation page

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204649160044068

### CODE REVIEW

**Review by nananathanh — COMMENTED** [review]
@SamuelBenkeCalabresi  Looking good, one thing I noticed was that the `Continue` button is not disabled when the mandatory opt-ins aren't selected.

File: apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx [hunk 1]
```diff
@@ -207,26 +227,34 @@ const CustomerForm = ({ customer, optIns, onSubmit }: CustomerFormProps) => {
...
+              <Heading level={2}>Payment Details</Heading>
+              <Box
+                direction={size === 'small' ? 'column' : 'row'}
+                align="start"
+                gap={size === 'small' ? 'none' : 'medium'}
+                justify="between"
+              >
+                <Box width={size === 'small' ? '100%' : '48%'}>
+                  <TextInputFormField
+                    name="country"
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx:239]: Wrong name here, means both inputs are always showing the same value

File: apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx [hunk 2]
```diff
@@ -84,6 +84,26 @@ const CustomerForm = ({ customer, optIns, onSubmit }: CustomerFormProps) => {
...
+                      <div
+                        dangerouslySetInnerHTML={{
+                          __html: `${optIn.text}`,
+                        }}
+                      />
+                      {optIn.mandatory && '*'}
+                    </Box>
+                  ),
+                  value: optIn.id,
+                  required: optIn.mandatory,
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx:101]: Not sure this is working on my end. 

Would required in this sense mean this needs a value? where not being checked counts as a value? Not sure on the implementation though. I wonder if you can do it with yup though.

File: apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx (hunk 1 above)
Comment (nananathanh) [reply apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx:239]: Is there a reason you need these inputs on this PR, you just remove them in the next?

File: apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx (hunk 2 above)
Comment (SamuelBenkeCalabresi) [reply apps/lineup-mercury/src/app/components/CustomerForm/CustomerForm.tsx:101]: I added a mandatory opt in and an optional one for event 186, and without the mandatory opt in doesn't allow to submit the value

**Pull request merged**